java/cuvs-java/src/main/java/com/nvidia/cuvs/CuVSDeviceMatrix.java

```diff
@@ -39,4 +30,67 @@ default CuVSHostMatrix toHost() {
     toHost(hostMatrix);
     return hostMatrix;
   }
+
+  default CuVSDeviceMatrix toDevice(CuVSResources resources) {
+    return new CuVSDeviceMatrixDelegate(this);
+  }
+
+  class CuVSDeviceMatrixDelegate implements CuVSDeviceMatrix {
+
+    @Override
+    public long size() {
+      return deviceMatrix.size();
+    }
+
+    @Override
+    public long columns() {
+      return deviceMatrix.columns();
+    }
+
+    @Override
+    public DataType dataType() {
+      return deviceMatrix.dataType();
+    }
+
+    @Override
+    public RowView getRow(long row) {
+      return deviceMatrix.getRow(row);
+    }
+
+    @Override
+    public void toArray(int[][] array) {
+      deviceMatrix.toArray(array);
+    }
+
+    @Override
+    public void toArray(float[][] array) {
+      deviceMatrix.toArray(array);
+    }
+
+    @Override
+    public void toArray(byte[][] array) {
+      deviceMatrix.toArray(array);
+    }
+
+    @Override
+    public void toHost(CuVSHostMatrix hostMatrix) {
+      deviceMatrix.toHost(hostMatrix);
+    }
+
+    @Override
+    public void toDevice(CuVSDeviceMatrix deviceMatrix, CuVSResources cuVSResources) {
+      this.deviceMatrix.toDevice(deviceMatrix, cuVSResources);
+    }
+
+    @Override
+    public void close() {
+      deviceMatrix.close();
```

Comment:
Wait, shouldn't this be a NOOP, like with `CuVSHostMatrixDelegate`?

Reply:
Good catch, you are 100% correct.